src/main/java/com/theelixrlabs/healthcare/service/PatientService.java

```diff
@@ -131,4 +132,30 @@ public String deletePatientById(String patientId) throws CustomException {
         patientRepository.deleteById(validPatientId);
         return messageUtil.getMessage(PatientConstants.PATIENT_DELETE_SUCCESS_MESSAGE, new Object[]{validPatientId});
     }
+
+    /**
+     * Retrieves a list of patients whose first or last name starts with the given name.
+     *
+     * @param patientName The name to search for. This should be a string representing
+     *                    the starting letters of the patient's first or last name.
+     * @return A list of PatientDTO objects representing the matching patients.
+     */
+    public List<PatientDto> getPatientsByName(String patientName) {
+        validator.validatePatientName(patientName);
+        List<PatientModel> patientModelList = patientRepository.searchByPatientName(patientName);
+        if (patientModelList.isEmpty()) {
+            throw new ResourceNotFoundException(messageUtil.getMessage(MessageConstants.PATIENT_NAME_NOT_FOUND));
```

Comment:
if data is not found for the given patientName, you can return the empty list right instead of sending an error message. or sending an error message is the requirement if data is not found?

Reply:
yes sending an error message is the requirement.